keepalived-vip/utils.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"net"
+	"os"
+	"regexp"
+	"sort"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	k8sexec "k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/sysctl"
+)
+
+const (
+	vethRegex = "^veth.*"
+)
+
+var (
+	invalidIfaces = []string{"lo", "docker0", "flannel.1"}
```

Comment:
Why not make these flags with defaults (or a list of invalid interfaces names as a single flag), since the docker interface can be setup as cbr0 by the kubelet?